build-conventions/src/main/java/org/elasticsearch/gradle/internal/checkstyle/HiddenFieldCheck.java

```diff
@@ -365,7 +365,8 @@ private static String capitalize(final String name) {
         // we should not capitalize the first character if the second
         // one is a capital one, since according to JavaBeans spec
         // setXYzz() is a setter for XYzz property, not for xYzz one.
-        if (name.length() == 1 || Character.isUpperCase(name.charAt(1)) == false) {
+        // @pugnascotia: unless the first char is 'x'.
```

Comment:
This allows a setter called `setXContentType` to have a parameter of `xContentType`.